internal/telemetry/metrics/providers.go

```diff
@@ -55,6 +55,9 @@ func newProxyMetricsHandler(promHandler http.Handler, envoyURL url.URL) http.Han
 	return func(w http.ResponseWriter, r *http.Request) {
 		defer promHandler.ServeHTTP(w, r)
 
+		// Ensure we don't get entangled with compression from ocprom
+		r.Header.Del("Accept-Encoding")
```

Comment:
Weird, but 🤷‍♂️ 👌

Reply:
Yeah...we do this, or we get into having to uncompress the response from the internal prometheus handler to add in our metrics.  It seems easier to just ensure it doesn't use compression.

Long term we should switch to accepting metrics via RPC and putting them into the local registry directly.  The current implementation is a bit of a hack and won't work with anything other than prometheus export format.